frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { Alert, FormGroup, Select, SelectOption } from '@patternfly/react-core';
+import { ProjectDetailsContext } from '~/pages/projects/ProjectDetailsContext';
+import { getDataConnectionDisplayName } from '~/pages/projects/screens/detail/data-connections/utils';
+
+type ExistingDataConnectionFieldProps = {
+  fieldId: string;
+  selectedDataConnection?: string;
+  setDataConnection: (name?: string) => void;
+  selectDirection?: 'up' | 'down';
+  menuAppendTo?: HTMLElement | 'parent';
```

Comment:
Could always just pick these -- no need to manually reimplement PF props. I suspect we don't do `Pick`s all that often, but it is something we can improve on in our component props